src/common/InputRowRadioButtons.js

```diff
@@ -0,0 +1,154 @@
+
+type Props<TItemKey> = $ReadOnly<{|
+  /**
+   * Component must be under the stack nav with the selectable-options screen.
+   *
+   * Pass this down from props or `useNavigation`.
+   */
+  navigation: AppNavigationProp<>,
+
+  /** What the setting is about, e.g., "Theme". */
+  label: string,
+
+  /** What the setting is about, in a short sentence or two. */
+  description?: string,
+
+  /** The current value of the input. */
+  valueKey: TItemKey,
+
+  /**
+   * The available options to choose from, with unique `key`s, one of which
+   * must be `valueKey`. items: $ReadOnlyArray<Item<TItemKey>>,
+   */
+  items: $ReadOnlyArray<Item<TItemKey>>,
+
+  onValueChange: (newValue: TItemKey) => void,
+|}>;
+
+/**
+ * A form-input row for the user to make a choice, radio-button style.
+ *
+ * Shows the current value (the selected item), represented as the item's
+ * `.title`. When tapped, opens the selectable-options screen, where the
+ * user can change the selection or read more about each selection.
+ */
+// This has the navigate-to-nested-screen semantics of NestedNavRow,
+// represented by IconRight. NestedNavRow would probably be the wrong
+// abstraction, though, because it isn't an imput component; it doesn't have
+// a value to display.
+export default function InputRowRadioButtons<TItemKey: string>(props: Props<TItemKey>): Node {
+  const { navigation, label, description, valueKey, items, onValueChange } = props;
+
+  const screenKey = useRef(
+    `selectable-options-${Math.floor(Math.random() * Number.MAX_SAFE_INTEGER).toString(36)}`,
+  ).current;
+
+  const selectedItem = items.find(c => c.key === valueKey);
+  invariant(selectedItem != null, 'InputRowRadioButtons: exactly one choice must be selected');
+
+  const screenParams = useMemo(
+    () => ({
+      title: label,
+      description,
+      items: items.map(({ key, title, subtitle }) => ({
+        key,
+        title,
+        subtitle,
+        selected: key === valueKey,
+      })),
+      onRequestSelectionChange: (itemKey, requestedValue) => {
+        if (!requestedValue || itemKey === valueKey) {
+          // Can't unselect a radio button.
+          return;
+        }
+        navigation.goBack();
+        onValueChange(itemKey);
+      },
+    }),
+    [navigation, label, description, items, valueKey, onValueChange],
+  );
+
+  const handleRowPressed = useCallback(() => {
+    // TODO: Can use .push instead of .navigate? See if Flow types are wrong
+    //   in saying we can't pass `key` when using .push.
+    navigation.navigate({
+      name: 'selectable-options',
+      key: screenKey,
+      params: screenParams,
+    });
+  }, [navigation, screenKey, screenParams]);
+
+  // Live-update the selectable-options screen.
+  useEffect(() => {
+    navigation.dispatch(
+      state =>
+        state.routes.find(route => route.key === screenKey)
+          ? { ...CommonActions.setParams(screenParams), source: screenKey }
+          : CommonActions.reset(state), // no-op
```

Comment:
Huh, funky. I'm reading about the relevant part of the React Nav API here:
https://reactnavigation.org/docs/5.x/navigation-prop#dispatch
https://reactnavigation.org/docs/5.x/navigation-actions/#setparams

Do we need the conditional and the no-op case? What happens if we just say:
```suggestion
  // Live-update the selectable-options screen.
  useEffect(() => {
    navigation.dispatch(
      { ...CommonActions.setParams(screenParams), source: screenKey },
```
; does that blow up if the route key passed as `source` isn't actually found on any route?

Reply:
On pressing the "Edit stream" button, which brings me to `EditStreamScreen`, I get this:

<img width=400 src="https://user-images.githubusercontent.com/22248748/167721061-742fca8c-ac49-4cbc-968f-9be08e4bd029.png" />

It's dismissable and probably harmless? Glad to have the dev-only warning, I guess, rather than a [silent failure](https://github.com/zulip/zulip-mobile/pull/3502#issuecomment-526765525), to help debug when one does expect `navigation.dispatch` to do something visible and instead nothing happens.

We can set `onUnhandledAction` on the navigation container, to set the behavior: https://reactnavigation.org/docs/navigation-container#onunhandledaction . That doc is for 6.x. The doc for 5.x, which we're on, doesn't mention the prop, but I'm seeing its implementation in `node_modules`. By default, it looks like [nothing happens in production](https://github.com/react-navigation/react-navigation/blob/%40react-navigation/core%405.16.1/packages/core/src/BaseNavigationContainer.tsx#L364).

Reply:
Cool. I think given that it's an error in dev, and it's not hard to write the code to avoid it, it's best to avoid it.